### PR TITLE
Added missing network details and methods

### LINKED ISSUE #273: Possible Omissions?
Hi please do correct me if I'm wrong :)

Should `GetNetworkAsync` in TVShows? not return the following?
https://developers.themoviedb.org/3/networks/get-network-details

Also these two methods seem to be missing as well for Networks?
https://developers.themoviedb.org/3/networks/get-network-alternative-names
https://developers.themoviedb.org/3/networks/get-network-images

Love the work you have done - works a treat :)

Kind regards

Daniel

### PR DESCRIPTION
**Pull request by FrederikBolding**
Fixes #273

### CODE REVIEW

File: TMDbLib/Objects/General/AlternativeNames.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public class AlternativeName
+    {
+        [JsonProperty("name")]
+        public string Name { get; set; }
+
+        [JsonProperty("type")]
+        public string Type { get; set; }
+    }
```
Comment (LordMike) [inline TMDbLib/Objects/General/AlternativeNames.cs]: Split these in two files, named by their class names.

File: TMDbLib/Objects/TvShows/NetworkBase.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Newtonsoft.Json;
+
+namespace TMDbLib.Objects.TvShows
+{
+    public class NetworkBase
```
Comment (LordMike) [inline TMDbLib/Objects/TvShows/NetworkBase.cs]: We only use this in one place?

File: TMDbLib/Objects/TvShows/TvShow.cs [hunk 3]
```diff
@@ -75,7 +75,7 @@ public class TvShow
         public string Name { get; set; }
 
         [JsonProperty("networks")]
-        public List<Network> Networks { get; set; }
+        public List<NetworkBase> Networks { get; set; }
```
Comment (LordMike) [inline TMDbLib/Objects/TvShows/TvShow.cs]: Don't we need a custom converter for this?

Deserializing to concrete types is a messy affair.

**Comment by FrederikBolding** [comment]
@LordMike I'm not exactly sure what you mean with regards to the network class changes. The idea was that the shared `NetworkBase` would be used for the methods that only return a few of the variables and the expanded one for the new methods.

**Comment by LordMike** [comment]
Ah yes - but when you deserialize, Newtonsoft is going to look at it and say "At this `Networks` property, I need a list of `NetworkBase`" and do exactly that. No magic is in place to provide you with the more concrete type of f.ex. `Network`.

I haven't run it yet. But I'm pretty confident you'll not get any `Network` instances out, but only instances of `NetworkBase`. You can verify by making `NetworkBase` abstract, and see if it still works (since newtonsoft can instantiate an abstract class).

**Comment by FrederikBolding** [comment]
Yes. And that was the idea since we needed an object that just contained the few variables returned when talking "generally" about networks. The detailed object then extends that, but the "general" info objects are `NetworkBase`.

**Comment by FrederikBolding** [comment]
Isn't that a good idea or would you rather that we do something else? - Just seemed like the simplest solution.

**Comment by LordMike** [comment]
Oh - it's because the list of networks only contains objects with those properties?..

I misunderstood it as the list would contain "one of" the `NetworkBase`.

**Comment by FrederikBolding** [comment]
Exactly 😄

**Pull request merged**